serviceregistry/src/main/java/eu/arrowhead/core/serviceregistry/database/service/ServiceRegistryDBService.java

```diff
@@ -1008,12 +1014,13 @@ public SystemResponseDTO getSystemByNameAndAddressAndPortResponse(final String s
 		
 		final int validatedPort = validateSystemPort(port);
 		final String validatedSystemName = validateSystemParamString(systemName);
-		final String validatedAddress = validateSystemParamString(address);
+		final String validatedAddress = networkAddressPreProcessor.normalize(address);
+		networkAddressVerifier.verify(validatedAddress);
 		
 		try {
 			final Optional<System> systemOptional = systemRepository.findBySystemNameAndAddressAndPort(validatedSystemName, validatedAddress, validatedPort);
 			if (systemOptional.isEmpty()) {
-				throw new InvalidParameterException("No system with name: " + systemName + ", address: " +  address + " and port: " + port);
+				throw new InvalidParameterException("No system with name: " + systemName + ", validatedSystemName: " +  validatedAddress + " and port: " + validatedPort);
```

Comment:
this exception is mismatch labels and values:
-systemName should be validatedSystemName,
-validatedSystemName should be address